sroka/api/mysql/mysql.py

```diff
@@ -0,0 +1,124 @@
+from configparser import NoOptionError, NoSectionError
+
+import os
+import mysql.connector
+from mysql.connector.errors import DatabaseError, OperationalError, InternalError
+import pandas as pd
+from retrying import retry
+
+import sroka.config.config as config
+
+
+@retry(stop_max_attempt_number=1,
+       wait_exponential_multiplier=1 * 2,
+       wait_exponential_max=1 * 2 * 2)
+def query_mysql(query: str, filename=None):
+    try:
+        options = get_options_from_config()
+
+    except NoSectionError:
+        print('Missing MySQL section in configuration')
+        return pd.DataFrame([])
+
+    if not validate_options(options):
+        return pd.DataFrame([])
+
+    try:
+        # Connect while passing only the arguments that were set in the
+        # configuration, since some are mutually exclusive or optional.
+        connection = mysql.connector.connect(**{k: v for k, v in options.items() if v != ""})
+
+        # Get the MySQL connector cursor, which allows interacting with the
+        # MySQL server.
+        cursor = connection.cursor()
+
+        # Execute the query.
+        cursor.execute(query)
+
+    except OperationalError as e:
+        print('Operational MySQL Error: {}'.format(e))
+        return pd.DataFrame([])
+    except InternalError as e:
+        print('Internal MySQL Error: {}'.format(e))
+        return pd.DataFrame([])
+    except DatabaseError as e:
+        print('Database MySQL Error: {}'.format(e))
+        return pd.DataFrame([])
+
+    # Cycle through the returned elements to build a pandas DataFrame.
+    df = pd.DataFrame(cursor, columns=cursor.column_names)
+
+    # Close the connection to the MySQL server.
+    cursor.close()
+    connection.close()
+
+    # Get path to folder in which to create the file.
+    filepath = "\\".join(filename.split('\\')[:-1])
+
+    # If it does not exist, create it.
+    if not os.path.exists(filepath):
```

Comment:
it will fail if user defines only name of file like `aaa.csv` because in this case filepath is empty

Reply:
it will fail in line 60, `os.makedirs("", exist_ok=True)`

Reply:
Will fix as soon as I have some time :p Thanks!